## Assignment (60 pts + 20 pts of Extra Credits) ##

**You should update the path below**

import pandas as pd
<br>
pathEmployees = r'C:\Users\f_sohnoume\OneDrive - UNCG\f_sohnoume (faculty)\Spring 2021\ISM210\Resources\Week 8\Employees.xlsx'
<br>
pathSuperstore = r'C:\Users\f_sohnoume\OneDrive - UNCG\f_sohnoume (faculty)\Spring 2021\ISM210\Resources\Week 8\Sample - Superstore.xlsx'
<br>
emloyeesData = pd.read_excel(pathEmployees)
<br>
superstoreData = pd.read_excel(pathSuperstore)



* Merge emloyeesData and superstoreData using EmployeeID ( 5 pts)
* Using superstoreData, calculate the average Sales per Region (10 pts)
    * Which Region has the highest average sales?
    * Which Region has the lowest average Sales?
* Using superstoreData, calculate the total Profit per Category (10 pts)
    * Which Category has the highest total Profit?
    * Which Category has the lowest total Profit?
* Using superstoreData, calculate the total Profit per City within State (15 pts)
    * Which City has the highest total Profit in California?
    * Which City has the highest total Profit in Florida?
    * How many Cities have a negative total Profit?
* Using superstoreData, calculate the total Quantity per Sub-Category within Region (10 pts)
    * Which Sub-Category has the highest total Quantity in the East region?
    * Which Sub-Category has the highest total Quantity in the South region?
* Using superstoreData, calculate the total Profit per State within Region (10 pts)
    * Which State has the highest total Profit in the East region?
    * Which State has the highest total Profit in the South region?
* Using superstoreData, calculate the average Sales per Region within Year of Order Date (10 pts)
    * Note: First, you will need to create a column Year of Order Date by calculating the Year of the Order Date. This is the code that returns the Year of the Order Date. **superstoreData['Order Date'].dt.year**
    * Which State has the highest total Profit in the East region in 2018?
    * Which State has the highest total Profit in the South region in 2017?
* Using superstoreData, calculate the total Profit per State within Year of Order Date (10 pts)
    * Which State has the highest total Profit in 2018?
    * Which State has the lowest total Profit in 2017?

In [1]:
import pandas as pd
pathEmployees = r'Employees.xlsx'
pathSuperstore = r'Sample - Superstore.xlsx'
emloyeesData = pd.read_excel(pathEmployees)
superstoreData = pd.read_excel(pathSuperstore)

In [2]:
emloyeesData.head()

EmployeeID   LastName FirstName                  Title TitleOfCourtesy  \
0           1    Davolio     Nancy   Sales Representative             Ms.   
1           2     Fuller    Andrew  Vice President, Sales             Dr.   
2           3  Leverling     Janet   Sales Representative             Ms.   
3           4    Peacock  Margaret   Sales Representative            Mrs.   
4           5   Buchanan    Steven          Sales Manager             Mr.   

   BirthDate   HireDate               Address      City PostalCode Country  \
0 1948-12-08 1992-05-01    507 - 20th Ave. E.   Seattle      98122     USA   
1 1952-02-19 1992-08-14    908 W. Capital Way    Tacoma      98401     USA   
2 1963-08-30 1992-04-01    722 Moss Bay Blvd.  Kirkland      98033     USA   
3 1937-09-19 1993-05-03  4110 Old Redmond Rd.   Redmond      98052     USA   
4 1955-03-04 1993-10-17       14 Garrett Hill    London    SW1 8JR      UK   

        HomePhone  Extension  
0  (206) 555-9857       5467  
1  (206) 555-9482       3457  
2  (206) 555-3412       3355  
3  (206) 555-8122       5176  
4   (71) 555-4848       3453

In [3]:
superstoreData.head()

Row ID        Order ID Order Date  Ship Date       Ship Mode Customer ID  \
0       1  CA-2017-152156 2017-11-08 2017-11-11    Second Class    CG-12520   
1       2  CA-2017-152156 2017-11-08 2017-11-11    Second Class    CG-12520   
2       3  CA-2017-138688 2017-06-12 2017-06-16    Second Class    DV-13045   
3       4  US-2016-108966 2016-10-11 2016-10-18  Standard Class    SO-20335   
4       5  US-2016-108966 2016-10-11 2016-10-18  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ... Region  \
0      Claire Gute   Consumer  United States        Henderson  ...  South   
1      Claire Gute   Consumer  United States        Henderson  ...  South   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   West   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...  South   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...  South   

        Product ID         Category Sub-Category  \
0  FUR-BO-10001798        Furniture    Bookcases   
1  FUR-CH-10000454        Furniture       Chairs   
2  OFF-LA-10000240  Office Supplies       Labels   
3  FUR-TA-10000577        Furniture       Tables   
4  OFF-ST-10000760  Office Supplies      Storage   

                                        Product Name     Sales Quantity  \
0                  Bush Somerset Collection Bookcase  261.9600        2   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400        3   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200        2   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775        5   
4                     Eldon Fold 'N Roll Cart System   22.3680        2   

   Discount    Profit  EmployeeID  
0      0.00   41.9136           5  
1      0.00  219.5820           1  
2      0.00    6.8714           7  
3      0.45 -383.0310           9  
4      0.20    2.5164           6  

[5 rows x 22 columns]

# Merge emloyeesData and superstoreData using EmployeeID ( 5 pts)

In [4]:
result = pd.merge(emloyeesData, superstoreData, left_on='EmployeeID', right_on='EmployeeID', how='left')
result

EmployeeID   LastName FirstName                 Title TitleOfCourtesy  \
0              1    Davolio     Nancy  Sales Representative             Ms.   
1              1    Davolio     Nancy  Sales Representative             Ms.   
2              1    Davolio     Nancy  Sales Representative             Ms.   
3              1    Davolio     Nancy  Sales Representative             Ms.   
4              1    Davolio     Nancy  Sales Representative             Ms.   
...          ...        ...       ...                   ...             ...   
9989           9  Dodsworth      Anne  Sales Representative             Ms.   
9990           9  Dodsworth      Anne  Sales Representative             Ms.   
9991           9  Dodsworth      Anne  Sales Representative             Ms.   
9992           9  Dodsworth      Anne  Sales Representative             Ms.   
9993           9  Dodsworth      Anne  Sales Representative             Ms.   

      BirthDate   HireDate             Address   City_x PostalCode  ...  \
0    1948-12-08 1992-05-01  507 - 20th Ave. E.  Seattle      98122  ...   
1    1948-12-08 1992-05-01  507 - 20th Ave. E.  Seattle      98122  ...   
2    1948-12-08 1992-05-01  507 - 20th Ave. E.  Seattle      98122  ...   
3    1948-12-08 1992-05-01  507 - 20th Ave. E.  Seattle      98122  ...   
4    1948-12-08 1992-05-01  507 - 20th Ave. E.  Seattle      98122  ...   
...         ...        ...                 ...      ...        ...  ...   
9989 1966-01-27 1994-11-15   7 Houndstooth Rd.   London    WG2 7LT  ...   
9990 1966-01-27 1994-11-15   7 Houndstooth Rd.   London    WG2 7LT  ...   
9991 1966-01-27 1994-11-15   7 Houndstooth Rd.   London    WG2 7LT  ...   
9992 1966-01-27 1994-11-15   7 Houndstooth Rd.   London    WG2 7LT  ...   
9993 1966-01-27 1994-11-15   7 Houndstooth Rd.   London    WG2 7LT  ...   

     Postal Code   Region       Product ID         Category Sub-Category  \
0        42420.0    South  FUR-CH-10000454        Furniture       Chairs   
1        53711.0  Central  OFF-ST-10004186  Office Supplies      Storage   
2        94109.0     West  TEC-PH-10001949       Technology       Phones   
3        94109.0     West  OFF-BI-10002215  Office Supplies      Binders   
4        68025.0  Central  OFF-AR-10000246  Office Supplies          Art   
...          ...      ...              ...              ...          ...   
9989     60653.0  Central  FUR-CH-10003379        Furniture       Chairs   
9990     10009.0     East  OFF-AP-10000595  Office Supplies   Appliances   
9991     45040.0     East  OFF-BI-10000494  Office Supplies      Binders   
9992      8701.0     East  OFF-AR-10002467  Office Supplies          Art   
9993     46203.0  Central  TEC-AC-10001714       Technology  Accessories   

                                           Product Name    Sales Quantity  \
0     Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.940        3   
1     Stur-D-Stor Shelving, Vertical 5-Shelf: 72"H x...  665.880        6   
2                               Cisco SPA 501G IP Phone  213.480        3   
3           Wilson Jones Hanging View Binder, White, 1"   22.720        4   
4                                            Newell 318   19.460        7   
...                                                 ...      ...      ...   
9989  Global Commerce Series High-Back Swivel/Tilt C...  797.944        4   
9990                 Disposable Triple-Filter Dust Bags   13.110        3   
9991       Acco Economy Flexible Poly Round Ring Binder    3.132        2   
9992                          Dixon Ticonderoga Pencils    8.940        3   
9993             Logitech MX Performance Wireless Mouse   39.890        1   

     Discount    Profit  
0         0.0  219.5820  
1         0.0   13.3176  
2         0.2   16.0110  
3         0.2    7.3840  
4         0.0    5.0596  
...       ...       ...  
9989      0.3  -56.9960  
9990      0.0    3.4086  
9991      0.7   -2.6100  
9992      0.0    2.4138  
9993      0.0   14.7593  

[9994

# Using superstoreData, calculate the average Sales per Region (10 pts)

In [5]:
superstoreData = superstoreData.reset_index(drop=True)
superstoreData = superstoreData.set_index('Region')
superstoreData.mean(level='Region').sort_values(by=['Profit'], ascending=False)

Row ID   Postal Code       Sales  Quantity  Discount     Profit  \
Region                                                                          
West     4986.524508  91707.696534  226.493233  3.829535  0.109335  33.849032   
East     4941.685042  17663.803313  238.336110  3.728230  0.145365  32.135808   
South    5069.935185  34490.772840  241.803645  3.832716  0.147253  28.857673   
Central  5030.547998  65340.572536  215.772661  3.779595  0.240353  17.092709   

         EmployeeID  
Region               
West       4.896972  
East       4.906601  
South      4.743827  
Central    4.889798

## Which Region has the highest average sales?

**West**

## Which Region has the lowest average Sales?

**Central**

# Using superstoreData, calculate the total Profit per Category (10 pts)

In [6]:
superstoreData1 = superstoreData.reset_index(drop=True)
superstoreData1 = superstoreData1.set_index('Category')
superstoreData1.sum(level='Category').sort_values(by=['Profit'], ascending=False)

Row ID  Postal Code        Sales  Quantity  Discount  \
Category                                                                  
Technology        9241154  102587530.0  836154.0330      6939    244.40   
Office Supplies  30010535  330740424.0  719047.0320     22906    947.80   
Furniture        10693326  118185210.0  741999.7953      8028    368.89   

                      Profit  EmployeeID  
Category                                  
Technology       145454.9481        9159  
Office Supplies  122490.8008       29255  
Furniture         18451.2728       10289

## Which Category has the highest total Profit?

**Technology**

## Which Category has the lowest total Profit?

**Furniture**

# Using superstoreData, calculate the total Profit per City within State (15 pts)

In [7]:
superstoreData2 = superstoreData.reset_index()
superstoreData2 = superstoreData2.set_index(['State','City'])
superstoreData2.sum(level=['State','City'])

Row ID  Postal Code        Sales  Quantity  \
State          City                                                           
Kentucky       Henderson         220967    2036160.0   15661.0100       185   
California     Los Angeles      3785302   67252887.0  175851.3410      2879   
Florida        Fort Lauderdale    81200     499665.0    4929.5325        65   
North Carolina Concord            43988     224216.0    5111.8440        44   
Washington     Seattle          2199185   41989758.0  119540.7420      1590   
...                                 ...          ...          ...       ...   
Arkansas       Springdale          9761      72762.0       4.3000         2   
California     Lodi               19652     190480.0     136.9800         9   
Texas          La Porte           19671     155142.0      30.3920         6   
Ohio           Mason              29610     135120.0      94.2640         7   
Georgia        Woodstock          19765      60376.0     286.6500        16   

                                Discount      Profit  EmployeeID  
State          City                                               
Kentucky       Henderson             0.0   4518.8169         231  
California     Los Angeles          55.5  30440.7579        3567  
Florida        Fort Lauderdale       4.5   -161.3551          87  
North Carolina Concord               1.8  -1788.6868          39  
Washington     Seattle              27.8  29156.0967        2151  
...                                  ...         ...         ...  
Arkansas       Springdale            0.0      1.4190           5  
California     Lodi                  0.0     44.4858          15  
Texas          La Porte              1.4    -32.5272           7  
Ohio           Mason                 1.6     -4.2352          17  
Georgia        Woodstock             0.0    104.8788           6  

[604 rows x 7 columns]

## Which City has the highest total Profit in California?

In [8]:
employeePerState2 = superstoreData2.sum(level=['State','City']).reset_index()
employeePerState2[employeePerState2.State == "California"].sort_values(by=['Profit'], ascending=False)

State             City   Row ID  Postal Code       Sales  Quantity  \
1    California      Los Angeles  3785302   67252887.0  175851.341      2879   
8    California    San Francisco  2462471   47998395.0  112669.092      1935   
73   California        San Diego   827518   15650880.0   47521.029       670   
351  California       Sacramento    85559    1245699.0    7311.278        36   
137  California          Anaheim   190906    2505708.0    7986.870       112   
..          ...              ...      ...          ...         ...       ...   
568  California      Morgan Hill    31642     380148.0     545.010        16   
551  California       Montebello     6835      90640.0      27.880         2   
597  California           Dublin     9672      94568.0      22.000         5   
598  California  San Luis Obispo     9702      93405.0       3.620         2   
320  California          Ontario     2072      91761.0     283.920         5   

     Discount      Profit  EmployeeID  
1       55.50  30440.7579        3567  
8       34.00  17507.3854        2487  
73      13.60   6377.1960         863  
351      1.60   2069.6859          55  
137      2.20   1234.0036         136  
..        ...         ...         ...  
568      0.35      7.8460          18  
551      0.00      3.9032           8  
597      0.20      1.3750           6  
598      0.00      1.1946           7  
320      0.20    -46.1370           2  

[81 rows x 9 columns]

**Los Angeles**

## Which City has the highest total Profit in Florida?


In [9]:
employeePerState2 = superstoreData2.sum(level=['State','City']).reset_index()
employeePerState2[employeePerState2.State == "Florida"].sort_values(by=['Profit'], ascending=False)

State              City  Row ID  Postal Code       Sales  Quantity  \
366  Florida       Tallahassee   96709     549151.0   7250.6690        89   
387  Florida           Orlando   77292     361229.0   1740.0800        39   
88   Florida  Saint Petersburg   96710     471940.0   1690.7920        53   
235  Florida           Deltona   15037     163625.0    893.2400        19   
447  Florida         Hollywood   79424     363231.0   1070.4740        44   
574  Florida         Kissimmee    8365      34741.0    751.9840         2   
303  Florida            Apopka   30330     228984.0    904.5540        31   
224  Florida     Boynton Beach   27409     334370.0    469.5850        28   
446  Florida           Miramar   52535     264184.0   1531.1920        41   
554  Florida   West Palm Beach   24031     100221.0    141.3280        11   
544  Florida      Delray Beach   25144     100335.0    242.5360        11   
584  Florida      Coral Gables   17845      66268.0     64.2480         6   
213  Florida         Homestead    9029      99090.0    132.5520         5   
222  Florida     Coral Springs   28922     264520.0    824.9920        21   
159  Florida        Palm Coast   19876     160685.0    274.4280        20   
383  Florida     Pompano Beach   13936      99204.0     90.4400        12   
425  Florida        Plantation   50947     366487.0   1436.8360        45   
15   Florida         Melbourne      44      32935.0     95.6160         2   
426  Florida  Port Saint Lucie   16264     104856.0    248.9500        11   
423  Florida       North Miami   12517      66322.0     22.1280         6   
297  Florida       Port Orange    1893      32127.0      7.8240         1   
465  Florida           Margate    4422      33063.0     15.5520         3   
508  Florida           Jupiter    5441      33458.0      2.0640         1   
487  Florida         Pensacola    4879      32503.0      2.2140         3   
325  Florida      Ormond Beach    2121      32174.0      2.8080         3   
578  Florida           Sanford   17129      65542.0    225.5460         9   
193  Florida        Boca Raton   16949     100299.0     63.0080         8   
459  Florida     Daytona Beach   25084     128456.0    272.3315        10   
101  Florida    Pembroke Pines   71449     462336.0   1714.3755        47   
2    Florida   Fort Lauderdale   81200     499665.0   4929.5325        65   
161  Florida           Hialeah  107418     594216.0   2338.9490        76   
132  Florida          Lakeland   97116     743622.0   5606.8865        74   
70   Florida             Tampa  152946    1210104.0   5196.0805       105   
80   Florida           Tamarac    7121     133276.0   1413.5100        15   
106  Florida             Miami  312221    1890806.0   8673.0745       215   
116  Florida      Jacksonville  348500    2416200.0  39133.3280       248   

     Discount     Profit  EmployeeID  
366      4.70   392.7116          85  
387      3.20   377.7253          46  
88       3.80   151.9823          69  
235      1.50   122.1344          21  
447      3.20   108.2012          61  
574      0.20    84.5982           9  
303      2.90    54.3634          34  
224      2.50    50.6615          41  
446      2.10    32.2810          36  
554      0.60    31.6092          17  
544      0.60    29.0542          19  
584      0.40    22.3578          10  
213      0.60    22.3291           4  
222      1.60    18.5568          29  
159      2.00    17.6552          26  
383      0.60    15.0065           7  
425      3.20    12.0391          50  
15       0.20     9.5616           3  
426      1.10     4.4490           5  
423      0.40     3.0798          13  
297      0.20     2.9340           3  
465      0.20     2.3328           7  
508      0.20     0.1548           8  
487      0.70    -1.4760           3  
325      0.70    -1.9656           3  
578      0.90   -11.8272           4  
193      1.10   -22.6456          16  
459      1.05   -52.3340          17  
101      4.55   -72.4491  

**Tallahassee**

## How many Cities have a negative total Profit?

In [10]:
employeePerState2[employeePerState2.Profit < 0]

State             City   Row ID  Postal Code        Sales  \
2           Florida  Fort Lauderdale    81200     499665.0    4929.5325   
3    North Carolina          Concord    43988     224216.0    5111.8440   
10     Pennsylvania     Philadelphia  2531454   10275302.0  109077.0130   
12            Texas          Houston  2017125   29052387.0   64504.7604   
13            Texas       Richardson       73     150160.0    1288.4640   
..              ...              ...      ...          ...          ...   
585        Illinois       Romeoville     8981      60441.0       8.9520   
588           Texas            Bryan    56743     466818.0     616.6560   
594        Illinois           Oswego    28440     181629.0     321.9780   
601           Texas         La Porte    19671     155142.0      30.3920   
602            Ohio            Mason    29610     135120.0      94.2640   

     Quantity  Discount      Profit  EmployeeID  
2          65      4.50   -161.3551          87  
3          44      1.80  -1788.6868          39  
10       1981    175.50 -13837.7674        2645  
12       1466    143.14 -10153.5485        1859  
13         12      0.80    -24.4893          10  
..        ...       ...         ...         ...  
585         2      0.80    -14.7708           6  
588        27      3.70   -213.7116          40  
594        13      1.80   -536.1276          11  
601         6      1.40    -32.5272           7  
602         7      1.60     -4.2352          17  

[138 rows x 9 columns]

**138 Cities has negative profits**

# Using superstoreData, calculate the total Quantity per Sub-Category within Region (10 pts)

In [11]:
superstoreData3 = superstoreData.reset_index()
superstoreData3 = superstoreData3.set_index(['Region','Sub-Category'])
superstoreData3 = superstoreData3.sum(level=['Region','Sub-Category']).reset_index()

## Which Sub-Category has the highest total Quantity in the East region?

In [12]:
superstoreData3[superstoreData3.Region == "East"].sort_values(by=['Quantity'], ascending=False)

Region Sub-Category   Row ID  Postal Code       Sales  Quantity  Discount  \
20   East      Binders  2223462    7914099.0   53497.997      1652     155.5   
36   East        Paper  1849553    6214121.0   20172.602      1400      24.4   
21   East  Furnishings  1353952    5063444.0   29071.380      1015      22.0   
32   East       Phones  1335723    4974678.0  100614.982       982      43.6   
35   East      Storage  1192579    4060063.0   71612.584       907      17.2   
23   East          Art  1067193    4109933.0    7485.764       849      15.8   
31   East  Accessories  1029635    3881235.0   45033.372       724      17.6   
17   East       Chairs   906491    2819441.0   96260.683       676      25.3   
60   East   Appliances   660836    2227398.0   34188.466       474       9.2   
53   East       Labels   530974    1781911.0    2602.934       420       7.2   
34   East    Fasteners   264592    1210079.0     819.718       298       5.6   
58   East       Tables   393127    1450415.0   39139.807       271      29.9   
22   East    Envelopes   345152    1369910.0    4375.874       267       6.0   
19   East    Bookcases   337583    1063550.0   43819.334       252      15.4   
51   East     Supplies   299209     871489.0   10760.116       195       3.8   
54   East     Machines   175855     730672.0   66106.165       165      10.5   
66   East      Copiers   108003     369772.0   53219.462        71       5.0   

        Profit  EmployeeID  
20  11267.9346        2199  
36   9015.3710        1866  
21   5881.4071        1364  
32  12314.6860        1315  
35   8389.3712        1185  
23   1899.9443        1083  
31  11195.8644        1034  
17   9357.7706         842  
60   8391.4134         549  
53   1129.2800         529  
34    263.9908         311  
58 -11025.3801         366  
22   1812.4090         385  
19  -1167.6318         348  
51  -1155.1352         284  
54   6928.6429         196  
66  17022.8418         118

**Binders**

## Which Sub-Category has the highest total Quantity in the South region?

In [13]:
superstoreData3[superstoreData3.Region == "South"].sort_values(by=['Quantity'], ascending=False)

Region Sub-Category   Row ID  Postal Code      Sales  Quantity  Discount  \
40  South      Binders  1290292    8629936.0  37030.341       981     92.40   
11  South        Paper  1143367    7709290.0  14150.984       851     22.60   
38  South  Furnishings   830545    5645992.0  17306.684       615     17.60   
44  South          Art   732234    4854182.0   4655.622       547     14.40   
43  South       Phones   683667    4673172.0  58304.438       526     14.00   
42  South  Accessories   665057    4825175.0  27276.754       504     11.00   
4   South      Storage   642126    4269310.0  35768.060       488     11.80   
1   South       Chairs   412365    3141701.0  45176.446       331      8.60   
39  South   Appliances   415872    2604773.0  19525.326       293      9.00   
3   South       Tables   262382    1691531.0  43916.192       227     11.35   
57  South       Labels   346755    2139034.0   2353.180       216      7.00   
41  South    Envelopes   229164    1897470.0   3345.556       206      4.00   
62  South    Fasteners   135371    1024376.0    503.316       126      2.40   
0   South    Bookcases   161276     992079.0  10899.362       118      2.80   
55  South     Supplies   144589     943802.0   8318.928        92      3.00   
63  South     Machines    78048     608388.0  53890.960        62      6.00   
67  South      Copiers    40185     224841.0   9299.756        26      0.60   

        Profit  EmployeeID  
40   3900.6640        1161  
11   5947.0614        1059  
38   3442.6829         813  
44   1058.5866         671  
43  10767.2753         704  
42   7004.5424         574  
4    2274.2965         616  
1    6612.0893         364  
39   4123.9396         399  
3   -4623.0579         244  
57   1040.7723         283  
41   1465.4770         272  
62    173.7181         144  
0    1339.4918         112  
55      1.8773         128  
63  -1438.8930          99  
67   3658.9067          42

**Binders**

# Using superstoreData, calculate the total Profit per State within Region (10 pts)


In [14]:
superstoreData4 = superstoreData.reset_index()
superstoreData4 = superstoreData4.set_index(['Region','Sub-Category'])
superstoreData4= superstoreData4.sum(level=['Region','Sub-Category']).reset_index()

## Which State has the highest total Profit in the East region?


In [15]:
superstoreData3[superstoreData3.Region == "East"].sort_values(by=['Profit'], ascending=False)

Region Sub-Category   Row ID  Postal Code       Sales  Quantity  Discount  \
66   East      Copiers   108003     369772.0   53219.462        71       5.0   
32   East       Phones  1335723    4974678.0  100614.982       982      43.6   
20   East      Binders  2223462    7914099.0   53497.997      1652     155.5   
31   East  Accessories  1029635    3881235.0   45033.372       724      17.6   
17   East       Chairs   906491    2819441.0   96260.683       676      25.3   
36   East        Paper  1849553    6214121.0   20172.602      1400      24.4   
60   East   Appliances   660836    2227398.0   34188.466       474       9.2   
35   East      Storage  1192579    4060063.0   71612.584       907      17.2   
54   East     Machines   175855     730672.0   66106.165       165      10.5   
21   East  Furnishings  1353952    5063444.0   29071.380      1015      22.0   
23   East          Art  1067193    4109933.0    7485.764       849      15.8   
22   East    Envelopes   345152    1369910.0    4375.874       267       6.0   
53   East       Labels   530974    1781911.0    2602.934       420       7.2   
34   East    Fasteners   264592    1210079.0     819.718       298       5.6   
51   East     Supplies   299209     871489.0   10760.116       195       3.8   
19   East    Bookcases   337583    1063550.0   43819.334       252      15.4   
58   East       Tables   393127    1450415.0   39139.807       271      29.9   

        Profit  EmployeeID  
66  17022.8418         118  
32  12314.6860        1315  
20  11267.9346        2199  
31  11195.8644        1034  
17   9357.7706         842  
36   9015.3710        1866  
60   8391.4134         549  
35   8389.3712        1185  
54   6928.6429         196  
21   5881.4071        1364  
23   1899.9443        1083  
22   1812.4090         385  
53   1129.2800         529  
34    263.9908         311  
51  -1155.1352         284  
19  -1167.6318         348  
58 -11025.3801         366

**Copiers**

## Which State has the highest total Profit in the South region?

In [16]:
superstoreData3[superstoreData3.Region == "South"].sort_values(by=['Profit'], ascending=False) 

Region Sub-Category   Row ID  Postal Code      Sales  Quantity  Discount  \
43  South       Phones   683667    4673172.0  58304.438       526     14.00   
42  South  Accessories   665057    4825175.0  27276.754       504     11.00   
1   South       Chairs   412365    3141701.0  45176.446       331      8.60   
11  South        Paper  1143367    7709290.0  14150.984       851     22.60   
39  South   Appliances   415872    2604773.0  19525.326       293      9.00   
40  South      Binders  1290292    8629936.0  37030.341       981     92.40   
67  South      Copiers    40185     224841.0   9299.756        26      0.60   
38  South  Furnishings   830545    5645992.0  17306.684       615     17.60   
4   South      Storage   642126    4269310.0  35768.060       488     11.80   
41  South    Envelopes   229164    1897470.0   3345.556       206      4.00   
0   South    Bookcases   161276     992079.0  10899.362       118      2.80   
44  South          Art   732234    4854182.0   4655.622       547     14.40   
57  South       Labels   346755    2139034.0   2353.180       216      7.00   
62  South    Fasteners   135371    1024376.0    503.316       126      2.40   
55  South     Supplies   144589     943802.0   8318.928        92      3.00   
63  South     Machines    78048     608388.0  53890.960        62      6.00   
3   South       Tables   262382    1691531.0  43916.192       227     11.35   

        Profit  EmployeeID  
43  10767.2753         704  
42   7004.5424         574  
1    6612.0893         364  
11   5947.0614        1059  
39   4123.9396         399  
40   3900.6640        1161  
67   3658.9067          42  
38   3442.6829         813  
4    2274.2965         616  
41   1465.4770         272  
0    1339.4918         112  
44   1058.5866         671  
57   1040.7723         283  
62    173.7181         144  
55      1.8773         128  
63  -1438.8930          99  
3   -4623.0579         244

**Phones**

# Using superstoreData, calculate the average Sales per Region within Year of Order Date (10 pts)
##Note: First, you will need to create a column Year of Order Date by calculating the Year of the Order Date. This is the code that returns the Year of the Order Date. superstoreData['Order Date'].dt.year


In [17]:
superstoreData5 = superstoreData.reset_index()
superstoreData5['Year'] = superstoreData5["Order Date"].dt.year
superstoreData5 = superstoreData5.set_index(['Year','Region'])
superstoreData5 = superstoreData5.sum(level=['Year','Region']).reset_index()

## Which State has the highest total Profit in the East region in 2018?

In [18]:
superstoreData5[superstoreData5.Year == 2018].sort_values(by=['Profit'], ascending=False)

Year   Region   Row ID  Postal Code        Sales  Quantity  Discount  \
13  2018     West  5596420  100813937.0  250128.3655      4270    115.20   
8   2018     East  4604152   16493327.0  213082.9040      3411    136.10   
4   2018    South  2803238   18274785.0  122905.8575      1915     80.50   
11  2018  Central  3844690   50480649.0  147098.1282      2880    186.42   

        Profit  EmployeeID  
13  43808.9561        5387  
8   33230.5614        4453  
4    8848.9079        2454  
11   7550.8442        3785

**West**

## Which State has the highest total Profit in the South region in 2017?

In [19]:
superstoreData5[superstoreData5.Year == 2017].sort_values(by=['Profit'], ascending=False)

Year   Region   Row ID  Postal Code        Sales  Quantity  Discount  \
1   2017     West  4084784   73958493.0  187480.1765      3025     94.35   
12  2017     East  3731373   13529374.0  180685.8220      2839    109.40   
7   2017  Central  2986885   39477394.0  147429.3760      2359    133.92   
0   2017    South  1975762   14016527.0   93610.2235      1614     62.65   

        Profit  EmployeeID  
1   24051.6069        3888  
12  20141.5961        3776  
7   19899.1629        2943  
0   17702.8084        1905

**West**

# Using superstoreData, calculate the total Profit per State within Year of Order Date (10 pts)


In [20]:
superstoreData6 = superstoreData.reset_index()
superstoreData6['Year'] = superstoreData6["Order Date"].dt.year
superstoreData6 = superstoreData6.set_index(['Year','State'])
superstoreData6 = superstoreData6.sum(level=['Year','State']).reset_index()

## Which State has the highest total Profit in 2018?


In [21]:
superstoreData6[superstoreData6.Year == 2018].sort_values(by=['Profit'], ascending=False)

Year                 State   Row ID  Postal Code        Sales  Quantity  \
36   2018            California  3431755   61239507.0  146388.3445      2633   
35   2018              New York  1568103    3643615.0   93922.9950      1319   
77   2018            Washington  1103192   21101318.0   65539.8960       806   
26   2018              Michigan   211889    3248248.0   25833.6490       258   
74   2018               Georgia   312456    1926136.0   19160.0200       235   
125  2018              Delaware   129863     375073.0   13754.9830        83   
82   2018               Indiana   336992    2863879.0   18516.4500       229   
78   2018              Kentucky   257338    2085518.0   15534.8400       186   
104  2018              Maryland   200898     780492.0    9451.6000       156   
32   2018             Minnesota   146488    2113181.0    6728.2500       138   
56   2018            New Jersey   255642     336813.0    9483.5500       140   
67   2018              Virginia   270050    1002150.0    7600.2000       162   
132  2018              Oklahoma    76672    1254007.0    6225.9100        64   
46   2018              Missouri   100378    1693706.0    9350.8000       113   
72   2018         Massachusetts   200959      80216.0    8143.0760       137   
94   2018             Wisconsin   177255    1760850.0    5567.4700       125   
146  2018           Connecticut   210498     222615.0    5307.2100       104   
86   2018               Montana     7222     297221.0    4229.2800        18   
136  2018             Louisiana    79057     987876.0    5501.5500        50   
79   2018           Mississippi    62332     508261.0    2997.2400        59   
152  2018              Arkansas   110224    1373856.0    2707.6400        78   
122  2018              Nebraska    52616     681342.0    3579.3500        31   
106  2018            New Mexico    76401    1927595.0    2818.5580        98   
115  2018               Alabama    80441     570606.0    1828.2500        57   
174  2018         New Hampshire    46289      22866.0    1509.4300        39   
150  2018                  Utah    68434    1264857.0    2460.6380        54   
76   2018          Rhode Island   122055      61084.0    3427.1100        75   
147  2018          South Dakota    26016     457122.0    1153.4100        32   
75   2018                Nevada    31211     534506.0    3138.1580        28   
31   2018        South Carolina    26162     235281.0    1560.7500        35   
95   2018                  Iowa    82611     822226.0     716.4100        51   
144  2018               Vermont    39789          0.0     842.2100        16   
156  2018                Kansas    31405     399906.0     733.1500        19   
16   2018               Florida   677157    4140781.0   26444.7155       414   
166  2018          North Dakota    36796     406721.0     919.9100        30   
154  2018                 Idaho    47049     667875.0    1233.5480        26   
177  2018         West Virginia    33175     104012.0    1209.8240        18   
133  2018  District of Columbia     3693      40032.0      77.7600         8   
34   2018                Oregon   150189    2921152.0    2886.5930       109   
43   2018               Arizona   354273    5967121.0   11133.5370       258   
44   2018                  Ohio   816741    7069912.0   23264.8500       603   
58   2018             Tennessee   537126    3057696.0   16113.8230       304   
89   2018              Colorado   326694    4892785.0   10299.8130       240   
4    2018        North Carolina   390895    2386624.0   23456.8290       335   
10   2018          Pennsylvania   976447    3756597.0   42688.3060       713   
15   2018              Illinois   918842   10449612.0   24351.6110       632   
13   2018                 Texas  1646730   24329849.0   43421.7582      1158   

     Discount      Profit  EmployeeID  
36      47.00  29366.4589        3141  
35      16.50  24357.0717        1661  
77      13.60  17256.7798        1123  
26  

**California**

## Which State has the lowest total Profit in 2017?

In [22]:
superstoreData6[superstoreData6.Year == 2017].sort_values(by=['Profit'], ascending=False)

Year                 State   Row ID  Postal Code        Sales  Quantity  \
1    2017            California  2754797   48406681.0  131551.9115      1999   
21   2017              New York  1545419    3356134.0   71844.1020      1193   
110  2017               Indiana   152525    1644158.0   25461.7800       134   
81   2017              Michigan   375611    3443756.0   27420.0350       275   
25   2017              Virginia   296912    1386081.0   26691.0500       254   
117  2017          Rhode Island    77443      43576.0   13085.6000        63   
62   2017            New Jersey   218282     375072.0   17983.3920       165   
5    2017            Washington   645153   11972587.0   19814.2840       451   
47   2017             Wisconsin   282493    2368344.0   13337.6200       186   
119  2017               Georgia   243072    1639296.0   14057.6700       199   
29   2017               Alabama    50707     689050.0    7651.3300        80   
113  2017                Nevada    49536     892020.0    9081.2720        46   
50   2017              Oklahoma   165707    2354281.0    8009.7900       122   
101  2017           Mississippi   105868     903592.0    5416.7800       111   
109  2017              Missouri    69212     916961.0    4221.1100        44   
0    2017              Kentucky   102275     909230.0    5102.8900        70   
80   2017         Massachusetts   124848      47778.0    3965.2640        82   
114  2017              Maryland   157558     525493.0    4709.5630        86   
19   2017              Delaware    93108     474104.0    2719.7200        90   
173  2017              Arkansas   132897    1446100.0    2224.1900        73   
55   2017           Connecticut    99037     143326.0    3998.4870        84   
9    2017              Nebraska    92288    1021323.0    3081.4200        73   
179  2017               Vermont    36243          0.0    2967.0600        26   
143  2017                  Utah    36016     673185.0    1822.4100        38   
17   2017             Minnesota    17017     442296.0    1527.3800        28   
39   2017                  Iowa    23079     408533.0     959.3100        31   
178  2017                 Idaho    35604     333004.0    1183.4860        13   
73   2017        North Carolina   343817    2075736.0   14967.2960       290   
131  2017                Kansas    29031     467334.0     691.0600        29   
103  2017         New Hampshire    19589      20084.0     933.5800        16   
145  2017                 Maine     4513       8802.0     547.3300         5   
135  2017        South Carolina    58770     321531.0     502.4800        42   
175  2017               Wyoming     5870      82001.0    1603.1360         4   
98   2017             Louisiana    26942     353042.0     262.7600        19   
128  2017  District of Columbia    11159     100080.0     117.0700        18   
51   2017            New Mexico    23311     352880.0      64.0800        11   
83   2017             Tennessee   183551    1463498.0    3130.4940       133   
64   2017               Florida   430951    2829371.0   13603.2835       343   
90   2017                Oregon   175942    3406989.0    5450.8370       135   
68   2017              Colorado   189616    4174274.0   10666.7410       172   
48   2017               Arizona   168939    3664872.0    6242.0190       156   
37   2017              Illinois   544122    7345700.0   21033.7190       437   
45   2017          Pennsylvania   703020    2628762.0   33066.6440       514   
14   2017                 Texas  1235800   19064708.0   41686.1520      1000   
65   2017                  Ohio   641154    5806163.0   24748.0100       497   

     Discount      Profit  EmployeeID  
1       40.45  20005.7161        2561  
21      19.90  16654.9495        1588  
110      0.00  10385.1339         187  
81       0.50   8992.5188         313  
25       0.00   7851.4730         267  
117      0.00   5160.0436          74  
62       0.60   4947.0245         232  
5   

**California**